### PR TITLE
Fix #787 : lowfi-landscape-audio-language-activity

### LINKED ISSUE #787: Lowfi Landscape: AudioLanguage Activity
Mock: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/85cbe1f0-8405-472b-abb0-fc83a0680aca/OP-Default-Audio-Language-/

Also, add test cases for this implementation.

### PR DESCRIPTION
**Pull request by chrk2205**
## Explanation
Fixes #787 
included a new color for RadioButton
Landscape versions for DefaultAudioActivity created and modified accordingly
Android Manifest changed accordingly

Mock: https://xd.adobe.com/view/ee9e607b-dbd6-4372-48dc-b687d32af3da-98af/screen/85cbe1f0-8405-472b-abb0-fc83a0680aca/OP-Default-Audio-Language-/
screenshot
WhatsApp Image 2020-03-15 at 00 25 51

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [ ] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@chrk2205 Please keep track of assignees section by following these guidelines.

https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section

File: app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt [hunk 1]
```diff
@@ -194,6 +196,48 @@ class OptionsFragmentTest {
...
+        )
+      onView(withContentDescription(R.string.go_to_previous_page)).perform(click())
+      onView(atPositionOnView(R.id.options_recyclerview, 2, R.id.audio_language_text_view)).check(
+        matches(withText("English"))
+      )
+    }
+  }
+
+    @Test
+  fun testOptionFragment_clickDefaultAudioLanguage_configurationChange_changeDefaultAudioLanguageToChineseSuccessfully() {
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: These two test cases are actually not helpful as you are not making any orientation change in between any clicks/selections from users.

You should write something like this:
`testOptionFragment_checkDefaultAudioLanguage_changeLanguageToHindi_changeConfiguration_checkHindiLanguageIsSelected`

**Review by rt4914 — CHANGES_REQUESTED** [review]
PTAL @chrk2205

File: app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt (hunk 1 above)
Comment (chrk2205) [reply app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: > @chrk2205 Please keep track of assignees section by following these guidelines.
> 
> https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section

@rt4914  sorry it was by miss it wont happen again!!

Comment (chrk2205) [reply app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: > These two test cases are actually not helpful as you are not making any orientation change in between any clicks/selections from users.
> 
> You should write something like this:
> `testOptionFragment_checkDefaultAudioLanguage_changeLanguageToHindi_changeConfiguration_checkHindiLanguageIsSelected`

  @Test
  fun testOptionFragment_checkDefaultAudioLanguage_changeLanguageToEnglish_changeConfiguration_checkEnglishLanguageIsSelected{
    launch<OptionsActivity>(createOptionActivityIntent(0)).use {
      onView(atPositionOnView(R.id.options_recyclerview, 2, R.id.audio_laguage_item_layout)).perform(
        click()
      )

      onView(withId(R.id.audio_language_recycler_view))
        .perform(
          RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(
            1,
            click()
          )
        )
      onView(withContentDescription(R.string.go_to_previous_page)).perform(click())
      onView(isRoot()).perform(orientationLandscape())
      onView(atPositionOnView(R.id.options_recyclerview, 2, R.id.audio_language_text_view)).check(
        matches(withText("English"))
      )
    }
  }

does this look good for that functtion @rt4914

**Comment by rt4914** [comment]
@chrk2205 Please use assignees section correctly as per guidelines mentioned in this link:
https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section

**Review by rt4914 — CHANGES_REQUESTED** [review]
Check this GIF:

options_issue


If I select any other language and do orientation change, the other language that I had selected disappears but it should remain selected. You will need to change implementation in `DefaultAudioLanguageActivity` to make sure that orientation change persists the state.

You will need to use the concept of `onSavedInstance`


Also, one important note, currently you have written test case even though the implementation in code is incorrect. So please in such cases make sure that you know that the implementation is correct or not and based on that decide whether to fix the implementation or not.

**Comment by chrk2205** [comment]
> If I select any other language and do orientation change, the other language that I had selected disappears but it should remain selected. You will need to change implementation in `DefaultAudioLanguageActivity` to make sure that orientation change persists the state.
> 
> You will need to use the concept of `onSavedInstance`
> 
> Also, one important note, currently you have written test case even though the implementation in code is incorrect. So please in such cases make sure that you know that the implementation is correct or not and based on that decide whether to fix the implementation or not.
 
i am not sure how to use it , is their any example or source  on how to use it ??
thanks.

**Comment by rt4914** [comment]
> > If I select any other language and do orientation change, the other language that I had selected disappears but it should remain selected. You will need to change implementation in `DefaultAudioLanguageActivity` to make sure that orientation change persists the state.
> > You will need to use the concept of `onSavedInstance`
> > Also, one important note, currently you have written test case even though the implementation in code is incorrect. So please in such cases make sure that you know that the implementation is correct or not and based on that decide whether to fix the implementation or not.
> 
> i am not sure how to use it , is their any example or source on how to use it ??
> thanks.

Reference: https://stackoverflow.com/questions/16769654/how-to-use-onsaveinstancestate-and-onrestoreinstancestate

Current code reference: `TopicLessonsFragment`

**Comment by chrk2205** [comment]
> Reference: https://stackoverflow.com/questions/16769654/how-to-use-onsaveinstancestate-and-onrestoreinstancestate
> 
> Current code reference: `TopicLessonsFragment`
I already know how onSavedInstance state work but not sure how to use in test cases!
I am supposed to modify Default audio Activity and then write the test case ??
Thanks

**Comment by chrk2205** [comment]
@rt4914 
I modified the DefaultAudioActivty rightnow the implementation is working properly !!

but i didnt wrote any test case in OptionsFragmentTest, should i write tests case for all the languages ??
Thanks

**Comment by rt4914** [comment]
> @rt4914
> I modified the DefaultAudioActivty rightnow the implementation is working properly !!
> 
> but i didnt wrote any test case in OptionsFragmentTest, should i write tests case for all the languages ??
> Thanks

@chrk2205 Please write test case. Not for all languages, just for 1-2 sample examples.

**Comment by chrk2205** [comment]
> @chrk2205 Please write test case. Not for all languages, just for 1-2 sample examples.

@rt4914  included 3 test cases .Hope the work is done
Thanks for helping me out.

File: app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt [hunk 2]
```diff
@@ -194,6 +196,66 @@ class OptionsFragmentTest {
...
 
+  @Test
+  fun testOptionFragment_checkDefaultAudioLanguage_changeLanguageToEnglish_changeConfiguration_checkEnglishLanguageIsSelected(){
+    launch<OptionsActivity>(createOptionActivityIntent(0)).use {
+      onView(atPositionOnView(R.id.options_recyclerview, 2, R.id.audio_laguage_item_layout)).perform(
+        click()
+      )
+      onView(withId(R.id.audio_language_recycler_view))
+        .perform(
+          RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: Nit change:
`actionOnItemAtPosition<RecyclerView.ViewHolder>(`

File: app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt [hunk 3]
```diff
@@ -194,6 +196,66 @@ class OptionsFragmentTest {
...
+
+  @Test
+  fun testOptionFragment_checkDefaultAudioLanguage_changeLanguageToChinese_changeConfiguration_checkChineseLanguageIsSelected(){
+    launch<OptionsActivity>(createOptionActivityIntent(0)).use {
+      onView(atPositionOnView(R.id.options_recyclerview, 2, R.id.audio_laguage_item_layout)).perform(
+        click()
+      )
+      onView(withId(R.id.audio_language_recycler_view))
+        .perform(
+          RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: Ditto as above

File: app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt [hunk 4]
```diff
@@ -194,6 +196,66 @@ class OptionsFragmentTest {
...
+
+  @Test
+  fun testOptionFragment_checkDefaultAudioLanguage_changeLanguageToHindi_changeConfiguration_checkHindiLanguageIsSelected(){
+    launch<OptionsActivity>(createOptionActivityIntent(0)).use {
+      onView(atPositionOnView(R.id.options_recyclerview, 2, R.id.audio_laguage_item_layout)).perform(
+        click()
+      )
+      onView(withId(R.id.audio_language_recycler_view))
+        .perform(
+          RecyclerViewActions.actionOnItemAtPosition<RecyclerView.ViewHolder>(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: Ditto as above

**Review by rt4914 — CHANGES_REQUESTED** [review]
Small change suggested.

File: app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt (hunk 4 above)
Comment (chrk2205) [reply app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: done

File: app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt (hunk 3 above)
Comment (chrk2205) [reply app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: done thanks

**Comment by chrk2205** [comment]
@rt4914  nit changes done thanks

File: app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt (hunk 2 above)
Comment (chrk2205) [reply app/src/sharedTest/java/org/oppia/app/options/OptionsFragmentTest.kt]: @rt4914 done thanks

**Review by rt4914 — CHANGES_REQUESTED** [review]
Actually the indentation in this file is not correct still.
Press `Control + Alt + L` - reformatting shortcut to fix the formatting.

**Review by rt4914 — APPROVED** [review]
LGTM, Thanks.

**Pull request merged**